src/views/TournamentRegister.vue

```diff
@@ -37,98 +37,192 @@ const register_form = reactive({
   team: '',
   pseudo: '',
   password: '',
-  role: '',
+  role: 'player',
   accept_rules: false,
 });
 
+const selected_team = computed(() => {
+  const res = (tournament.value?.teams as Team[]).filter((team) => team.name === register_form.team);
+  if (res.length === 1) {
+    return res[0];
+  }
+  return null;
+});
+
 const rules = computed(() => ({
   team: { required },
-  pseudo: { required },
-  password: { required },
+  pseudo: register_form.role === 'player' ? { required } : {},
+  password: { required, minLengthValue: minLength(8) },
   role: { required },
   accept_rules: { acceptRules },
 }));
 
 const v$ = useVuelidate(rules, register_form, { $autoDirty: true });
 
-const register_team = () => {
-  /* empty */
+const open_modal = ref(false);
+
+const register_team = async () => {
+  const isValid = await v$.value.$validate();
+  if (!isValid) return;
+
+  try {
+    await registerTeam(
+      tournament.value?.id as number,
+      register_form.team,
+      register_form.password,
+      register_form.role,
+      register_form.pseudo,
+      user.id,
+    );
+  } catch (err) {
+    return;
+  }
+
+  open_modal.value = true;
+};
+
+const register_player = async () => {
+  const isValid = await v$.value.$validate();
+  if (!isValid) return;
+
+  try {
+    await registerPlayerOrManager(
+      selected_team.value?.id as number,
+      register_form.pseudo,
+      register_form.password,
+      register_form.role,
+      tournament.value?.id as number,
+    );
+  } catch {
+    return;
+  }
+
+  open_modal.value = true;
+};
+
+const payment = async () => {
+  /* */
 };
 
-const register_player = () => {
-  /* empty */
+const generate_password = () => {
+  const symboles = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
```

Comment:
pretty sure there is more fancy way to do so but lgfm

Reply:
There doesn't seem to be a more obvious way to do this. We'll leave it at that for now.